code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -246,6 +246,10 @@
 
 // Please do not add one-off mob AIs here, but override this function for your mob
 /mob/living/simple_animal/hostile/CanAttack(atom/the_target)//Can we actually attack a possible target?
+	if(!isatom(the_target))
+		. = FALSE
+		CRASH("Invalid target: [the_target]")
```

Comment:
```suggestion
		stack_trace("Invalid target in CanAttack(): [the_target]")
		return FALSE
```

`stack_trace` does the same thing as `CRASH` but it actually continues the rest of the proc so let's do it like this rather than rely on the funny dot operator